numpy/lib/function_base.py

```diff
@@ -1320,6 +1320,477 @@ def gradient(f, *varargs, **kwargs):
+            # 1D equivalent -- out[0] = (y[1] - y[0])
+            out[slice1] = (y[slice2] - y[slice3])
+            
+            slice1[axis] = -1
+            slice2[axis] = -1
+            slice3[axis] = -2
+            # 1D equivalent -- out[-1] = (y[-1] - y[-2])
+            out[slice1] = (y[slice2] - y[slice3])
+            
+            # Numerical differentiation: 2st order edges, 2nd order interior
+        else:
+    	    # Use second order differences where possible
+            out = np.empty_like(y, dtype=otype)
+            
+            slice1[axis] = slice(1, -1)
+            slice2[axis] = slice(2, None)
+            slice3[axis] = slice(None, -2)
+            # 1D equivalent -- out[1:-1] = (y[2:] - y[:-2])/2.0
+            out[slice1] = (y[slice2] - y[slice3])/2.0
+            
+            slice1[axis] = 0
+            slice2[axis] = 0
+            slice3[axis] = 1
+            slice4[axis] = 2
+            # 1D equivalent -- out[0] = -(3*y[0] - 4*y[1] y[2]) / 2.0
+            out[slice1] = -(3.0*y[slice2] - 4.0*y[slice3] + y[slice4])/2.0
+            
+            slice1[axis] = -1
+            slice2[axis] = -1
+            slice3[axis] = -2
+            slice4[axis] = -3
+            # 1D equivalent -- out[-1] = (3*y[-1] - 4*y[-2] y[-3])
+            out[slice1] = (3.0*y[slice2] - 4.0*y[slice3] + y[slice4])/2.0
+        
+        # divide by step size
+        out /= dx[axis]
+        
+        if i == 0:
+            out0 = out
+        elif i == 1:
+            out1 = out
+        elif i == 2:
+            out2 = out
+        elif i == 3:
+            out3 = out
+        elif i == 4:
+            out4 = out
+        else:
+            out5 = out
+        
+        # reset the slice object in this dimension to ":"
+        slice1[axis] = slice(None)
+        slice2[axis] = slice(None)
+        slice3[axis] = slice(None)
+        slice4[axis] = slice(None)
+
+    
+    outvals[0] = out0 - out1
+    outvals[1] = out2 - out3
+    outvals[2] = out4 - out5
+    return outvals
+
+
+def laplace_s(f):
+    """
+    Return the laplacian of an N-dimensional array.
+
+    Computes the laplacian of an N-dimensional numpy array, as a
+    concatenation of the numpy functions gradient and divergence. 
+    For more information on each of these, type help(gradient) or 
+    help(divergence) 
+    
+    Parameters
+    ----------
+    f : array_like
+        Input array
+    	
+    Returns
+    -------
+    l : array_like
+        Laplacian of input array
+    	
+    See Also
+    --------
+    gradient, divergence
+    
+    Example
+    -------
+    This example returns an array that grows linearly in the X axis
+    after applying the laplacian function to the array X**3+Y**2+Z**2:
+    
+    >>> X,Y,Z=np.mgrid[0:200,0:200,0:200]
+    >>> f=X**3+Y**2+Z**2
+    >>> d=laplace_s(f)
+    >>> print np.shape(d)
+    >>> plt.imshow(d[:,:,1])
+    >>> plt.colorbar()
+    >>> plt.show()
+    """
+    
+    l = np.divergence(np.gradient(f))
```

Comment:
While I know this is mathematically sound, I am not sure that this is numerically very good or efficient?

I don't mind having a reasonable good and simple default laplacian in numpy, but I have a bit doubts that this is a good approach. Unfortunatly the numpy correlation function does not have a "mirror" mode, which would make a (typical?) stencil implementation very easy (more similar to the scipy one).